### PR TITLE
Docs sprites

### PR DESCRIPTION
**Pull request by pathunstrom**

### CODE REVIEW

File: ppb/sprites.py [hunk 1]
```diff
@@ -302,12 +435,16 @@ class Sprite(SquareShapeMixin, RenderableMixin, RotatableMixin, BaseSprite):
...
+    your game objects.
 
-    * BaseSprite
-    * SquareShapeMixin
-    * RenderableMixin
-    * RotatableMixin
+    If you need rectangular sprites instead of squares, see
+    :class:`RectangleSprite`.
 
     New in 0.7.0: Use this in place of BaseSprite in your games.
```
Comment (astronouth7303) [inline ppb/sprites.py:449]: Can we nix this?

File: ppb/sprites.py [hunk 2]
```diff
@@ -45,6 +54,30 @@ class BaseSprite:
     layer: int = 0
 
     def __init__(self, **kwargs):
+        """
+        :class:`BaseSprite` does not accept any positional arguments, and uses
+        kwargs to set arbitrary state to the :class:`BaseSprite` instance. This
```
Comment (astronouth7303) [inline ppb/sprites.py:59]: Is "kwargs" too jargony?

File: ppb/sprites.py [hunk 3]
```diff
@@ -64,8 +97,13 @@ class RenderableMixin:
     """
     A class implementing the API expected by ppb.systems.renderer.Renderer.
 
-    You should include RenderableMixin before BaseSprite in your parent
-    class definitions.
+    The render expects a width and height (see :class:`RectangleMixin`) and will
+    skip rendering if a sprite has no shape. You can use
+    :class:`RectangleMixin`, :class:`SquareMixin`, or set the values yourself.
+
+    Additionally, if :attr:`~RenderableMixin.image` is None, the sprite will not
```
Comment (astronouth7303) [inline ppb/sprites.py:104]:
```suggestion
    Additionally, if :attr:`~RenderableMixin.image` is ``None``, the sprite will not
```

File: ppb/sprites.py [hunk 4]
```diff
@@ -95,7 +133,10 @@ def __image__(self):
 
 class RotatableMixin:
     """
-    A simple rotation mixin. Can be included with sprites.
+    A rotation mixin. Can be included with sprites.
+
+    .. warning:: rotation does not affect underlying position, it only rotates
```
Comment (astronouth7303) [inline ppb/sprites.py:138]: "position" is the wrong word. Like, of course it doesn't affect the position.

File: ppb/sprites.py [hunk 5]
```diff
@@ -139,17 +182,27 @@ class RectangleShapeMixin:
 
     Classes derived from RectangleShapeMixin default to the same size and
     shape as all ppb Sprites: A 1 game unit by 1 game unit square. Just set
-    the width and height in your constructor (Or as class attributes) to
-    change this default.
+    the width and height in your constructor (Or as
+    :class:`class attributes <BaseSprite>`) to change this default.
+
+    .. warning:: The concrete class using RectangleShapeMixin must have a
```
Comment (astronouth7303) [inline ppb/sprites.py:188]:
```suggestion
    .. warning:: The concrete class using :class:`RectangleShapeMixin` must have a
```

Comment (astronouth7303) [inline ppb/sprites.py:188]: A warning seems a little bit dire for this? I also feel like we should note that this is provided by the default stuff.

File: ppb/sprites.py [hunk 6]
```diff
@@ -139,17 +182,27 @@ class RectangleShapeMixin:
...
+       position.
```
Comment (astronouth7303) [inline ppb/sprites.py:189]:
```suggestion
       ``position`` attribute.
```

File: ppb/sprites.py [hunk 7]
```diff
@@ -139,17 +182,27 @@ class RectangleShapeMixin:
...
+    #: The height of the sprite.
     height: int = 1
     # Following class properties for type hinting only. Your concrete sprite
     # should already have one.
     position: Vector
 
     @property
     def left(self) -> float:
+        """
+        The left side side of the object.
```
Comment (astronouth7303) [inline ppb/sprites.py:202]:
```suggestion
        The x-axis position of the left side of the object.
```

File: ppb/sprites.py [hunk 8]
```diff
@@ -158,6 +211,11 @@ def left(self, value: Union[float, int]):
 
     @property
     def right(self) -> float:
+        """
+        The right side of the object.
```
Comment (astronouth7303) [inline ppb/sprites.py:215]:
```suggestion
        The x-axis position of the right side of the object.
```

File: ppb/sprites.py [hunk 9]
```diff
@@ -166,6 +224,11 @@ def right(self, value: Union[float, int]):
 
     @property
     def top(self) -> float:
+        """
+        The top of the object.
```
Comment (astronouth7303) [inline ppb/sprites.py:228]:
```suggestion
        The y-axis position of the top of the object.
```

File: ppb/sprites.py [hunk 10]
```diff
@@ -174,6 +237,11 @@ def top(self, value: Union[int, float]):
 
     @property
     def bottom(self) -> float:
+        """
+        The bottom of the object.
```
Comment (astronouth7303) [inline ppb/sprites.py:241]:
```suggestion
        The y-axis position of the bottom of the object.
```

File: ppb/sprites.py [hunk 11]
```diff
@@ -182,6 +250,11 @@ def bottom(self, value: Union[float, int]):
 
     @property
     def top_left(self) -> Vector:
+        """
+        The top left corner of the object.
```
Comment (astronouth7303) [inline ppb/sprites.py:254]:
```suggestion
        The coordinates of the top left corner of the object.
```

File: ppb/sprites.py [hunk 12]
```diff
@@ -182,6 +250,11 @@ def bottom(self, value: Union[float, int]):
...
+
+        Can be set to a :class:`ppb_vector.Vector`.
```
Comment (astronouth7303) [inline ppb/sprites.py:256]: Is this where we want to point this?

File: ppb/sprites.py [hunk 13]
```diff
@@ -193,6 +266,11 @@ def top_left(self, vector: Vector):
 
     @property
     def top_right(self) -> Vector:
+        """
+        The top right corner of the object.
```
Comment (astronouth7303) [inline ppb/sprites.py:270]:
```suggestion
        The coordinates of the top right corner of the object.
```

File: ppb/sprites.py [hunk 14]
```diff
@@ -204,6 +282,11 @@ def top_right(self, vector: Vector):
 
     @property
     def bottom_left(self) -> Vector:
+        """
+        The bottom left corner of the object.
```
Comment (astronouth7303) [inline ppb/sprites.py:286]:
```suggestion
        The coordinates of the bottom left corner of the object.
```

File: ppb/sprites.py [hunk 15]
```diff
@@ -219,13 +302,23 @@ def bottom_right(self) -> Vector:
 
     @bottom_right.setter
     def bottom_right(self, vector: Vector):
+        """
+        The bottom right corner of the object.
```
Comment (astronouth7303) [inline ppb/sprites.py:306]:
```suggestion
        The coordinates of the bottom right corner of the object.
```

File: ppb/sprites.py [hunk 16]
```diff
@@ -219,13 +302,23 @@ def bottom_right(self) -> Vector:
...
+        """
         vector = Vector(vector)
         x = vector.x - (self.width / 2)
         y = vector.y + (self.height / 2)
         self.position = Vector(x, y)
 
     @property
     def bottom_middle(self) -> Vector:
+        """
+        The midpoint of the bottom of the object.
```
Comment (astronouth7303) [inline ppb/sprites.py:318]:
```suggestion
        The coordinates of the midpoint of the bottom of the object.
```

File: ppb/sprites.py [hunk 17]
```diff
@@ -235,6 +328,11 @@ def bottom_middle(self, value: VectorLike):
 
     @property
     def left_middle(self) -> Vector:
+        """
+        The midpoint of the left side of the object.
```
Comment (astronouth7303) [inline ppb/sprites.py:332]:
```suggestion
        The coordinates of the midpoint of the left side of the object.
```

File: ppb/sprites.py [hunk 18]
```diff
@@ -244,6 +342,11 @@ def left_middle(self, value: VectorLike):
 
     @property
     def right_middle(self) -> Vector:
+        """
+        The midpoint of the right side of the object.
```
Comment (astronouth7303) [inline ppb/sprites.py:346]:
```suggestion
        The coordinates of the midpoint of the right side of the object.
```

File: ppb/sprites.py [hunk 19]
```diff
@@ -253,6 +356,11 @@ def right_middle(self, value: VectorLike):
 
     @property
     def top_middle(self) -> Vector:
+        """
+        The midpoint of the top of the object.
```
Comment (astronouth7303) [inline ppb/sprites.py:360]:
```suggestion
        The coordinates of the midpoint of the top of the object.
```

File: ppb/sprites.py [hunk 20]
```diff
@@ -262,6 +370,12 @@ def top_middle(self, value: VectorLike):
 
     @property
     def center(self) -> Vector:
+        """
+        The center point of the object. Equivalent to the
```
Comment (astronouth7303) [inline ppb/sprites.py:374]:
```suggestion
        The coordinates of the center point of the object. Equivalent to the
```

File: ppb/sprites.py [hunk 21]
```diff
@@ -317,14 +454,8 @@ class RectangleSprite(RectangleShapeMixin, RenderableMixin, RotatableMixin, Base
...
-    * RectangleShapeMixin
-    * RenderableMixin
-    * RotatableMixin
-
-    .. warning::
-       The rotation and facing attributes of RectangleSprite do not change its
-       bounding box, only the presentation of the image applied to it.
+    Similarly to :class:`~ppb.Sprite`, :class:`RectangleSprite` does not
+    introduce any new methods or attributes. It's made up of :class:`BaseSprite`
+    with the mixins: :class:`RotatableMixin`, :class:`RenderableMixin`, and
```
Comment (astronouth7303) [inline ppb/sprites.py:459]:
```suggestion
    with the mixins :class:`RotatableMixin`, :class:`RenderableMixin`, and
```

File: ppb/sprites.py (hunk 5 above)
Comment (pathunstrom) [inline ppb/sprites.py:188]:
```suggestion
    .. note:: The concrete class using RectangleShapeMixin must have a
```

File: ppb/sprites.py (hunk 12 above)
Comment (pathunstrom) [reply ppb/sprites.py:256]: It's good enough for now and we can come back and change it if we change it.

File: ppb/sprites.py (hunk 4 above)
Comment (pathunstrom) [inline ppb/sprites.py:138]:
```suggestion
    .. warning:: rotation does not affect underlying shape (the corners are still in the same place), it only rotates
```

Comment (pathunstrom) [reply ppb/sprites.py:138]: How's this?

File: ppb/sprites.py (hunk 5 above)
Comment (pathunstrom) [reply ppb/sprites.py:188]: Going to modify this line to change the warning to a note so I can commit them together.

File: ppb/sprites.py (hunk 4 above)
Comment (astronouth7303) [reply ppb/sprites.py:138]: Sure

**Pull request merged**